OpenFileIO.md

```diff
@@ -0,0 +1,105 @@
+# "OpenFileIO" Proposal
+
+## Abstract
+
+This proposal calls for the creation of a minimal, open source C library
```

Comment:
Yeah def, just a lot of people have already said "will it have a C interface", I figured we can build all the rest on it?

Reply:
As you say - I wasn't going to put any implementation details in, it just had been asked about enough it felt like it might short circuit some fears that it would be bloatware?

Reply:
@feltech I updated the intro to speak of C and C++.